src/vs/workbench/services/preferences/browser/preferencesService.ts

```diff
@@ -593,6 +597,29 @@ export class PreferencesService extends Disposable implements IPreferencesServic
 		return position;
 	}
 
+	/**
+	 * Should be of the format:
+	 * 	code://settings/settingName
+	 * Examples:
+	 * 	code://settings/files.autoSave
+	 *
+	 */
+	async handleURL(uri: URI): Promise<boolean> {
+		if (uri.authority !== SETTINGS_AUTHORITY) {
+			return false;
+		}
+
+		const openSettingsOptions: IOpenSettingsOptions = {};
+		const settingInfo = uri.path.split('/').filter(part => !!part);
+		if (settingInfo.length === 0) {
+			return false;
+		}
+		openSettingsOptions.query = settingInfo[0];
```

Comment:
@rzhao271, is there a good way that I can check if `settingInfo` is actually a real setting?